Xcode 13 support + iOS 15 fixes

leandroalonso opened the pull request:
Adds support to Xcode 13 (including CI) and fixes issues for iOS 15 + Xcode 13 combo.

Included issues:

* #17225
* #17227
* #17233
* #17379

### To test

1. Run the app
2. Make sure the app has the same look 'n feel
3. Make sure CI is 🟢 

## Regression Notes
1. Potential unintended areas of impact
The whole project

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

leandroalonso committed:
FIx: tab bar background on iOS 15

leandroalonso committed:
Add: sets iOS 15 new sectionHeaderTopPadding to zero

leandroalonso committed:
Merge pull request #17225 from wordpress-mobile/issue/17172_fix_tab_bar_background

iOS 15: fix tab bar background

leandroalonso committed:
Merge pull request #17227 from wordpress-mobile/issue/ios_15_table_view_header_top_padding

iOS 15: fix the top padding on Table View's header

leandroalonso committed:
Change: Xcode image from 12 to 13

leandroalonso committed:
Fix: image name

mokagio committed:
Bump CircleCI Xcode image version to `13.0.0`

mokagio committed:
Bump `.xcversion` value from `12.5` to `13.0`

mokagio committed:
Use latest Ruby version in CircleCI, `2.7.4`

Prior to this change, we were using `2.6.6` but that version is no
longer available in the Xcode 13 image.

See error message here:
https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/24560/workflows/74788ee1-f42e-4876-96c6-4e75e74016d4/jobs/57379

Commit:
Remove MockContext from unit tests

Commit:
Rearrange project files

Commit:
Replace the use of MockContext with TestContextManager.mainContext in unit tests, to address build issues on Xcode 13

mokagio committed:
Update Simulator device and iOS version to use in CircleCI

mokagio committed:
Update Fastlane from 2.184.1 to 2.195.0

I did this to see if it solved the issues we are experiencing with Xcode
13 testing, but that didn't help.

ScoutHarris committed:
Merge branch 'develop' into issue/xcode_13

ScoutHarris committed:
Fix sheet being fullscreen in iOS15.

mokagio committed:
Use new Automattic Xcode 13 image in Buildkite

mokagio committed:
Run tests on iOS 15 in Buildkite

ScoutHarris committed:
Merge pull request #17379 from wordpress-mobile/fix/17378-user_profile_sheet_ios15

iOS 15: Fix user profile sheet being shown fullscreen

leandroalonso committed:
fastlane: filter tests to prevent test crashes

oguzkocer committed:
Update translations

oguzkocer committed:
Update metadata translations

oguzkocer committed:
Update Jetpack metadata translations

oguzkocer committed:
Bump version number

jkmassel committed:
Fix Ruby Issue

oguzkocer committed:
Update translations

oguzkocer committed:
Bump version number

oguzkocer committed:
Remove subtitle.txt files from jetpack_metadata

ScoutHarris committed:
Merge branch 'develop' into issue/xcode_13

ScoutHarris committed:
Merge branch 'develop' into issue/xcode_13

leandroalonso committed:
Merge branch 'issue/xcode_13' into issue/xcode_13_change_buildkite_image

mokagio committed:
Add note with rationale for skipping some tests in Fastlane

leandroalonso committed:
Merge pull request #17233 from wordpress-mobile/issue/xcode_13_change_buildkite_image

iOS 15: Change Buildkite image to Xcode 13

leandroalonso committed:
Merge branch 'develop' into issue/xcode_13

jkmassel reviewed (approved):
LGTM from the code side of things

mokagio committed:
Read Ruby version for chruby from .ruby-version in CircleCI

This ensures there is a single source of truth fro the Ruby version
across:

- Local tooling
- CircleCI
- Buildkite

mokagio committed:
Bump Ruby version to latest 2.x stable: 2.7.4

mokagio reviewed (commented):
@jkmassel @leandroalonso LGTM, too. I opened a PR against this branch to ensure we use the same Ruby version across the board: #17418. I think we should wait till that one is merged in here before merging into `develop`.

AliSoftware committed:
Merge pull request #17418 from wordpress-mobile/ruby-version-source-of-truth

Read Ruby version for chruby from .ruby-version in CircleCI

AliSoftware commented on .buildkite/commands/build-for-testing.sh:
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash -eu
 
+# Temporary fix until we're on the Xcode 13.1 VM
```
I fear that since those comments are hidden in the separate `.sh` files and thus not noticable directly from the `.yml` pipelines, we might miss them next time we update the image.

What do you think about doing something like this:

1. Add some hashtag to these comments, like:
```suggestion
# FIXIT-13.1: Temporary fix until we're on the Xcode 13.1 VM
```
2. Add a reference to those comment in the 2 `.yml` files:
```yml
    IMAGE_ID: xcode-13 # Remember to address all 'FIXIT-13.1' comments when we'll update to new image
```

AliSoftware reviewed (approved):
I've now merged #17418, so this should now be ready to merge.

I left a comment suggestion to ensure we won't forget to remove the "temporary fixes until Xcode 13.1" next time we update the `IMAGE_ID` (because I feel we'd otherwise miss it); take it or leave it 🙃

jkmassel committed:
Update .buildkite/commands/build-for-testing.sh

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

jkmassel committed:
Add fixit note

mokagio commented:
We sent out an internal communication (ref. pbArwn-38I) announcing the official Xcode 13 support and will merge this PR on November 9, just to give folks a couple of days to read the announcement and download Xcode 13 if they haven't done so yet.
